ls1intum/Artemis
# [Enhancement] - Allow score type change in quiz re-evaluation

### Checklist

### Motivation and Context


Sometimes instructors may want to change the way the quiz score is calculated into the course / exam score after releasing the exercise. This is not possible at the moment

### Description

Added the option to define the score calculation type in the re-evaluation of a quiz exercises.
Changes the wording from "course / exam" into either "exam" or "course", depending on if the exercises in in a course or exam.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create quiz (While creating, make sure that the score calculation as mentioned in the screenshots says "in the course score calculation")
4. conduct the quiz
5. After the quiz has ended, click on re-evaluate
6. make sure it is possible to change the score calculation type and that is saves correctly after clicking save at the bottom
7. navigate into the creation page of every exercise type and make sure that it says "Should this exercise be included in the course score calculation?"
8. Go into the exercise group page of a exam and go to the creation page of every exercise type
9. Check that it now says "Should this exercise be included in the **exam** score calculation?"


### Screenshots



Re-evaluation view before:
<img width="1057" alt="Screenshot 2021-05-22 at 11 54 23" src="https://user-images.githubusercontent.com/44401560/119222633-91c08800-baf5-11eb-9858-1a8d0c8d490d.png">

Re-evaluation view after:
<img width="913" alt="Screenshot 2021-05-22 at 11 53 20" src="https://user-images.githubusercontent.com/44401560/119222644-9e44e080-baf5-11eb-8c1b-8a3d70ec0f97.png">

Wording before:
![Screenshot 2021-05-22 at 12 03 52](https://user-images.githubusercontent.com/44401560/119222681-cfbdac00-baf5-11eb-8714-0edae12fbe8f.png)
Wording after:
<img width="381" alt="Screenshot 2021-05-22 at 11 52 43" src="https://user-images.githubusercontent.com/44401560/119222713-fb409680-baf5-11eb-8d9e-b29fdefaf7e4.png">
